plugins/c8y_configuration_plugin/src/main.rs

```diff
@@ -143,3 +148,27 @@ fn init(cfg_dir: PathBuf) -> Result<(), anyhow::Error> {
 fn mqtt_config(tedge_config: &TEdgeConfig) -> Result<MqttConfig, MqttConfigBuildError> {
     tedge_config.mqtt_config()
 }
+
+fn validate_config_dir_and_file(config_dir: &Path) -> Result<(), anyhow::Error> {
+    match config_dir.parent() {
+        Some(cdir) => {
+            if !Path::new(cdir).is_dir() {
+                anyhow::bail!(
+                    "The configuration directory {} does not exists",
+                    cdir.display()
+                );
+            } else {
+                if !Path::new(config_dir).is_file() {
+                    anyhow::bail!(
+                        "The configuration file {} does not exists",
+                        config_dir.display()
+                    );
+                }
```

Comment:
Why not just that check?

Reply:
There might be some times the `directory` exists but not the config file. So, wanted to be specific.

Reply:
We are both incorrect ;-) After some checks, it appears that the directory must exist but not necessarily the configuration file.

See https://github.com/thin-edge/thin-edge.io/blob/main/crates/extensions/c8y_config_manager/src/plugin_config.rs#L93

Reply:
Yes but it's okay to have no configuration : the defaults will apply. So one needs to check only the directory.